### ISSUE TITLE
Where is your innovation?

### ISSUE THREAD

**Issue by cj-jake** [kostas1515/FRACAL#3]
You put a lot of other things in there. Where is your focus? Where is your innovation? I can't find the innovation in your coder.

**Comment by kostas1515** [comment]
Hello, thank you for your interest in our work. The innovation is in the statistical calculation of the fractal dimension, which is explicitly shown in the ReadMe. If you read the "Optional - Get Dataset statistics" section, you will be able to calculate the fractal dimension of the classes for your dataset. After having these fractal weights you can perform logit adjustment as shown in the "\mmdet\models\roi_heads\bbox_heads\fracal_bbox_head.py". Thanks for your comment.

**Comment by cj-jake** [comment]
Thank you for your answer.

**Comment by cj-jake** [comment]
Hello, author. I’m back again with another question after reading your paper — thank you for the insightful work.

I was trying to better understand your point about the classes’ dependence on the location distribution pₛ(y, u).

If I understand correctly, this refers to certain classes tending to appear in specific regions of an image — for example (just my own understanding), hats might often appear at the top.

My question is: if such location-dependent characteristics do not exist, would the method still be applicable? What if the object can appear at completely random positions?

**Comment by kostas1515** [comment]
Hello and thank you for the question. 

The method works for both objects that show location-class independence and location-class dependence.  For the first case, FRACAL assigns the maximum fractal weight which is 2, and for the latter case it assigns the smallest fractal weight 1.  In our experiments we inversely downweight the softmax probabilities based on the fractal weights to balance the predictions. 

To come back to your question, if an object appears at completely random positions, FRACAL assigns a weight of 2. If you have a dataset with all classes that appear randomly and uniformly all across image locations, then you don't need to use FRACAL (If you use it then all classes  get down-weighted using the same factor i.e. 2 , and nothing changes in the softmax). If you have a dataset with classes that appear both uniformly and non-uniformly in the image space then FRACAL can give you a better performance trade-off, by our proposed logit re-balancing.
